### PR TITLE
[iOS] Update implementation related to bars' design

### LINKED ISSUE #483: [iOS] Use SwiftUI-Introspect
## Kind (Required)

Proposal / Discussion

## Overview (Required)

I propose to use [SwiftUI-Introspect](https://github.com/siteline/SwiftUI-Introspect) in iOS project.

Firstly, I need it to set `searchController` for #450.

Secondly, the implementation of `NavigationBarConfigurator` is a little risky for now.
It seems like designed for the similar purpose of `navigarionController?.navigarionBar` (current navigation stack), but the implementation is using `UINavigationBar.appearance()` (throughout the whole app), and will recover the previous settings when the view disappeared (like `navigationItem`).
And it has an unexpected behavior to recover the previous settings when push a new view, though it may not be a problem with the current design. 

With SwiftUI-Introspect, we should be able to achieve everything we want effortlessly.

### Throughout the whole app

```swift
UINavigationBar.appearance()
```

### Current navigation stack

```swift
.introspectNavigationController { navigationController in
    navigationController.navigationBar
}
```

### Current view controller

```swift
.introspectViewController { viewController in
    viewController.navigarionItem
}
```

If it's fine to use SwiftUI-Introspect, I would like to submit a PR using it to update the current implementation for `UINavigationBar` 🙋‍♂️

 ## Links	

- https://github.com/siteline/SwiftUI-Introspect

### PR DESCRIPTION
**Pull request by myihsan**
## Issue

- close #483 

## Overview (Required)

- Update bars appearances
- Implement `InlineTitleNavigationBarScrollView` (may need a better name) to achieve the design with [SwiftUI-Introspect](https://github.com/siteline/SwiftUI-Introspect)
- Update some screen for the changes above

## Screenshot

It's not completely the same as the design, but it looks like what the designer wanted according to `Home/First View/Scroll`, plus there is no way to change the effect's color in my experience. If there is a way, please tell me.
- [x] Confirm with the designer

### `InlineTitleNavigationBarScrollView`

Before | After
:--: | :--:
 | 
 | 

### Large Title

Before | After
:--: | :--:
 | 
 | 
 | 
 |

### CODE REVIEW

File: ios/Sources/Component/InlineTitleNavigationBarScrollView.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            .onPreferenceChange(ContentOffsetPreferenceKey.self) { value in
+                shouldApplyScrollEdgeAppearance = value <= 0
+            }
+            .introspectViewController { viewController in
+                guard let navigationBar = viewController.navigationController?.navigationBar else { return }
+                let navigationItem = viewController.navigationItem
+                navigationItem.standardAppearance = shouldApplyScrollEdgeAppearance
+                    ? navigationBar.scrollEdgeAppearance
+                    : navigationBar.standardAppearance
+                navigationBar.layoutIfNeeded()
```
Comment (myihsan) [inline ios/Sources/Component/InlineTitleNavigationBarScrollView.swift:50]: Planed to change gently following pan gesture using `UIViewPropertyAnimator`, but there is no property to animate but change the appearance and `layoutIfNeeded()` which will lose the progress when exterior reasons case `navigarionBar` re-layout.

Comment (myihsan) [reply ios/Sources/Component/InlineTitleNavigationBarScrollView.swift:50]: > Planed to change gently following pan gesture

This is the default behavior when build with Xcode 13, so we should remove `InlineTitleNavigationBarScrollView` when migrating to Xcode 13.

File: ios/Sources/SettingFeature/SettingToggleItem.swift [hunk 2]
```diff
@@ -16,7 +16,6 @@ public struct SettingToggleItem: View {
             Text(title)
         })
         .toggleStyle(SwitchToggleStyle(tint: AssetColor.primary.color))
-        .listRowBackground(AssetColor.Background.contents.color)
```
Comment (ry-itto) [inline ios/Sources/SettingFeature/SettingToggleItem.swift:19]: In design, this should be here. Please revert this change :pray:

In this PR|Design
--|--
スクリーンショット 2021-06-08 23 18 56|スクリーンショット 2021-06-08 23 18 36

File: ios/Sources/Component/InlineTitleNavigationBarScrollView.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public init(
+        showsIndicators: Bool = true,
+        @ViewBuilder content: () -> Content
+    ) {
+        self.showsIndicators = showsIndicators
+        self.content = content()
+    }
+
+    public var body: some View {
+        let _ = shouldApplyScrollEdgeAppearance
```
Comment (ry-itto) [inline ios/Sources/Component/InlineTitleNavigationBarScrollView.swift:30]: What is this property?
It seems not used 👀 

There are some TODOs?

File: ios/Sources/SettingFeature/SettingToggleItem.swift (hunk 2 above)
Comment (myihsan) [reply ios/Sources/SettingFeature/SettingToggleItem.swift:19]: Oh, I thought it was something for the `UITableView`.😱
I will fix it.

File: ios/Sources/Component/InlineTitleNavigationBarScrollView.swift (hunk 3 above)
Comment (myihsan) [reply ios/Sources/Component/InlineTitleNavigationBarScrollView.swift:30]: This line is required to aware for the change of `shouldApplyScrollEdgeAppearance`.

File: ios/Sources/SettingFeature/SettingToggleItem.swift (hunk 2 above)
Comment (ry-itto) [reply ios/Sources/SettingFeature/SettingToggleItem.swift:19]: No, it is not needed! Current state is okey.

**Review by ry-itto — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**